src/target/riscv/riscv.c

```diff
@@ -1622,8 +1691,9 @@ static int riscv_trigger_detect_hit_bits(struct target *target, int64_t *unique_
 		/* FIXME: this logic needs to be changed to ignore triggers that are not
 		 * the last one in the chain. */
 		if (tdata1 & hit_mask) {
-			LOG_TARGET_DEBUG(target, "Trigger %u (unique_id=%" PRIi64 ") has hit bit set.",
-				i, r->trigger_unique_id[i]);
+			LOG_TARGET_DEBUG(target, "Trigger %u (unique_id=%" PRIi64 ""
+			", need_single_step=%" PRId32 ") has hit bit set.",
+				i, r->trigger_unique_id[i], r->need_single_step);
```

Comment:
Cosmetic suggestions:

- The message would be more clear if "need_single_step" is mentioned at the end, not right next to "trigger" and "unique_id".
- PRId32 should not be used for printing of a bool. Instead, you can print strings "yes", "no".

```suggestion
			LOG_TARGET_DEBUG(target, "Trigger %u (unique_id=%" PRIi64 
			") has hit bit set. (need_single_step=%s)",
				i, r->trigger_unique_id[i], r->need_single_step ? "yes" : "no");
```

Reply:
addressed